src/Aspire.Hosting/Cli/CliOrphanDetector.cs

```diff
@@ -0,0 +1,51 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Hosting;
+
+namespace Aspire.Hosting.Cli;
+
+internal sealed class CliOrphanDetector(IConfiguration configuration, IHostApplicationLifetime lifetime) : BackgroundService
+{
+    private const string CliProcessIdEnvironmentVariable = "ASPIRE_CLI_PID";
+
+    internal TimeProvider TimeProvider { get; set; } = TimeProvider.System;
```

Comment:
Why not inject it?

Reply:
Will do ... I thought that we did it this way in other areas but I was wrong. It was also a similar approach to IsRunning just setting an internal property.